marlowe-runtime/sync-api/Language/Marlowe/Protocol/Query/Types.hs

```diff
@@ -111,6 +125,53 @@ data Page a b = Page
   deriving stock (Eq, Show, Read, Ord, Functor, Generic, Foldable, Traversable)
   deriving anyclass (ToJSON, Binary)
 
+data SomeContractState = forall v. SomeContractState (MarloweVersion v) (ContractState v)
+
+instance Show SomeContractState where
+  showsPrec p (SomeContractState MarloweV1 state) = showParen (p >= 11)
+    ( showString "SomeContractState"
+    . showSpace
+    . showsPrec 11 MarloweV1
+    . showSpace
+    . showsPrec 11 state
+    )
+
+instance Eq SomeContractState where
+  SomeContractState v state == SomeContractState v' state' = case testEquality v v' of
+    Nothing -> False
+    Just Refl -> case v of
+      MarloweV1 -> state == state'
+
+instance Binary SomeContractState where
+  put (SomeContractState MarloweV1 state) = do
+    put $ SomeMarloweVersion MarloweV1
+    put state
+  get = do
+    SomeMarloweVersion MarloweV1 <- get
+    SomeContractState MarloweV1 <$> get
+
+instance ToJSON SomeContractState where
+  toJSON (SomeContractState MarloweV1 state) = object
+    [ "version" .= MarloweV1
+    , "state" .= state
+    ]
+
+data ContractState v = ContractState
+  { contractId :: ContractId
+  , roleTokenMintingPolicyId :: PolicyId
+  , metadata :: TransactionMetadata
+  , initialBlock :: BlockHeader
+  , initialOutput :: TransactionScriptOutput v
+  , latestBlock :: BlockHeader
+  , latestOutput :: Maybe (TransactionScriptOutput v)
+  , availablePayouts :: Map TxOutRef (Payout v)
```

Comment:
Is `availablePayouts` the best terminology? Would `unclaimedPayouts` or `unspentPayouts` be better?

Reply:
Let's decide to adopt the terminology of `claimed` vs `unclaimed`.